app/admin/site_data_infos.rb

```diff
@@ -1,18 +1,148 @@
-  #
-  # permit_params :url_id, :test_id, :cms_type, :cms_version, :js, :cloudflare, :login_url, :host_name, :plugins, :themes
-  #
-  # or
-  #
-  # permit_params do
-  #   permitted = [:url_id, :test_id, :cms_type, :cms_version, :js, :cloudflare, :login_url, :host_name, :plugins, :themes]
-  #   permitted << :other if params[:action] == 'create' && current_user.admin?
-  #   permitted
-  # end
-  
+  actions :index, :show
+  filter :test_id
+  filter :url_id
+  filter :cloudflare, :as => :select, :collection => SiteDataInfo::CLOUDFLARESTATUS.invert
+  filter :id
+
+  scope :all, :default => true
+  scope :Plugins_found, :default => true do |site_data_infos|
+    site_data_infos.where.not(:plugins => '[]')    
+  end
+  scope :Themes_found, :default => true do |site_data_infos|
+    site_data_infos.where.not(:themes => '[]')
+  end
+  scope :Js_found, :default => true do |site_data_infos|
+    site_data_infos.where.not(:js => '[]')
+  end
+  scope :login_url_found, :default => true do |site_data_infos|
+    site_data_infos.where.not(:login_url => '0')
+  end
+  scope :WP_version_found, :default => true do |site_data_infos|
+    site_data_infos.where.not(:cms_version => 'version not found')
+  end
+
+  index do 
+    id_column
+    column "Test" do |site_data|
+      link_to site_data.test.id, admin_test_path(site_data.test)
+    end
+    column "Url" do |site_data|
+      link_to "#{site_data.url.id}:: #{site_data.url.url}", admin_url_path(site_data.url)
+    end 
+    column 'Plugins' do |site_data|
+      plugins = site_data.plugins
+      if JSON::parse(plugins).size > 0
+        link_to 'Plugins', admin_plugins_path("q[url_id_equals]" => site_data.url_id)
+      else
+        div("Not found", style: "color: red")
+      end
+    end
+    column 'Themes' do |site_data|
+      themes = site_data.themes
+      if JSON::parse(themes).size > 0
+        link_to 'Themes', admin_themes_path("q[url_id_equals]" => site_data.url_id)
+      else
+        div("Not found", style: "color: red")
+      end
+    end
+    column 'JS' do |site_data|
+      js = site_data.js
+      if JSON::parse(js).size > 0
+        link_to 'JS', admin_js_infos_path('q[url_id_equals]' => site_data.url_id)
+      else
+        div("Not found", style: "color: red")
+      end
+    end
+    column 'WP Version' do |site_data|
+      version = site_data.cms_version
+      if version == 'version not found'
+        div ('Not found'), :style => "color : red"
+      else
+        version
+      end
+    end
+    column 'Cloudflare' do |site|
+      if site.cloudflare == SiteDataInfo::CloudFlareStatus::INACTIVE
+        div (SiteDataInfo::CLOUDFLARESTATUS[site.cloudflare]),style: "color: red"
+      elsif site.cloudflare == SiteDataInfo::CloudFlareStatus::ACTIVE
+        div (SiteDataInfo::CLOUDFLARESTATUS[site.cloudflare]),style: "color: green"
+      end
+    end
+    column 'Login Url' do |site_data|
+      if site_data.login_url == SiteDataInfo::LoginUrl::NOTFOUND
```

Comment:
again with login url, it can simply be a string field, if it contains a string then it means that the string is the login URL, if it is nil, then that means you weren't able to find the login URL